.github/workflows/build.yml

```diff
@@ -0,0 +1,29 @@
+name: Build
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  goreleaser-build:
+    name: Build opkssh with GoReleaser
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - name: Run GoReleaser
+        uses: goreleaser/goreleaser-action@v6
+        with:
+          distribution: goreleaser
+          version: "~> v2"
+          args: release --clean --snapshot
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
What do I need to do to set this up?

Reply:
No needs to setup this secret, it is automatically defined when starting a new pipeline: https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication. The permission of the token is defined with the `permission` stanza.

For this file, the token is not needed as we don't generate a release on GitHub. I will remove it.